### PR TITLE
Room Settings: Name, Topic, Photo, Aliases, History Visibility

### PR DESCRIPTION
**Pull request by onurays**
WhatsApp Image 2020-06-24 at 11 18 15 (3)

WhatsApp Image 2020-06-24 at 11 18 15 (2)

WhatsApp Image 2020-06-24 at 11 20 14

WhatsApp Image 2020-06-24 at 11 18 15

WhatsApp Image 2020-06-24 at 11 18 15 (1)

### CODE REVIEW

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/state/DefaultStateService.kt [hunk 1]
```diff
@@ -84,4 +101,67 @@ internal class DefaultStateService @AssistedInject constructor(@Assisted private
...
+                stateKey = null
+        )
+    }
+
+    override fun updateAvatar(avatarUri: Uri, fileName: String, callback: MatrixCallback<Unit>): Cancelable {
+        val cancelableBag = CancelableBag()
+        val workerParams = UploadAvatarWorker.Params(sessionId, avatarUri, fileName)
+        val workerData = WorkerParamsFactory.toData(workerParams)
+
+        val uploadAvatarWork = workManagerProvider.matrixOneTimeWorkRequestBuilder<UploadAvatarWorker>()
```
Comment (ganfra) [inline matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/state/DefaultStateService.kt:137]: Same remark than the other PR with uploading avatar, don't think we should use a worker overthere

File: vector/src/main/java/im/vector/riotx/features/media/BigImageViewerActivity.kt [hunk 2]
```diff
@@ -57,14 +76,122 @@ class BigImageViewerActivity : VectorBaseActivity() {
...
+            AvatarSelectorView.Type.GALLERY -> {
+                MultiPicker.get(MultiPicker.IMAGE).single().startWith(this)
+            }
+        }
+    }
+
+    private fun onRoomAvatarSelected(image: MultiPickerImageType) {
+        val destinationFile = File(cacheDir, "${image.displayName}_edited_image_${System.currentTimeMillis()}")
+        val uri = image.contentUri
+        UCrop.of(uri, destinationFile.toUri())
```
Comment (ganfra) [inline vector/src/main/java/im/vector/riotx/features/media/BigImageViewerActivity.kt:118]: Ucrop again, create helper please

File: vector/src/main/java/im/vector/riotx/features/roomprofile/AvatarSelectorView.kt [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+import im.vector.riotx.features.roomprofile.AvatarSelectorView.Callback
+import kotlin.math.max
+
+private const val ANIMATION_DURATION = 250
+
+/**
+ * This class is the view presenting choices for picking avatar.
+ * It will return result through [Callback].
+ */
+class AvatarSelectorView(context: Context,
```
Comment (ganfra) [inline vector/src/main/java/im/vector/riotx/features/roomprofile/AvatarSelectorView.kt:55]: Why creating again this class, please refact the existing one to be more dynamic, or use a simple dialog?

File: vector/src/main/java/im/vector/riotx/core/epoxy/profiles/ProfileActionItem.kt [hunk 4]
```diff
@@ -63,7 +71,7 @@ abstract class ProfileActionItem : VectorEpoxyModel<ProfileActionItem.Holder>()
     override fun bind(holder: Holder) {
         super.bind(holder)
         holder.view.setOnClickListener(listener)
-        if (listener == null) {
+        if (listener == null || !editable) {
```
Comment (ganfra) [inline vector/src/main/java/im/vector/riotx/core/epoxy/profiles/ProfileActionItem.kt:74]: Why adding this? editable bool is for showing/hiding icon. Maybe it's not the good vocabulary, I think it's the one used by Nad, but not sure.

**Comment by bmarty** [comment]
Please fix this reported error:
```
❌ Error: 'android\.app\.AlertDialog' detected 1 time, in file './vector/src/main/java/im/vector/riotx/features/media/BigImageViewerActivity.kt', at line 20.
```

File: vector/src/main/java/im/vector/riotx/features/roomprofile/RoomProfileViewModel.kt [hunk 5]
```diff
@@ -111,11 +113,13 @@ class RoomProfileViewModel @AssistedInject constructor(@Assisted private val ini
...
-                }, {
-                    _viewEvents.post(RoomProfileViewEvents.Failure(it))
-                })
+                .subscribe(
+                        {
+                            _viewEvents.post(RoomProfileViewEvents.OnChangeAvatarSuccess)
+                        },
+                        {
+                            _viewEvents.post(RoomProfileViewEvents.Failure(it))
+                        })
```
Comment (onurays) [inline vector/src/main/java/im/vector/riotx/features/roomprofile/RoomProfileViewModel.kt:122]: We may put the last parentheses to the new line? :)

Comment (bmarty) [reply vector/src/main/java/im/vector/riotx/features/roomprofile/RoomProfileViewModel.kt]: Done

**Pull request merged**